elfi/examples/lorenz.py

```diff
@@ -0,0 +1,341 @@
+
+import numpy as np
+
+import elfi
+
+
+def _lorenz_ode(y, params):
+    """Parametrized Lorenz 96 system defined by a coupled stochastic differential equations (SDE).
+
+    Parameters
+    ----------
+    y : numpy.ndarray of dimension (batch_size, n_obs)
+        Current state of the SDE.
+    params : list
+        The list of parameters needed to evaluate function. In this case it is
+        list of four elements - eta, theta1, theta2 and f.
+
+    Returns
+    -------
+    dy_dt : np.array
+        Rate of change of the SDE.
+
+    """
+    dy_dt = np.empty_like(y)
+
+    eta = params[0]
+    theta1 = params[1]
+    theta2 = params[2]
+
+    f = params[3]
+
+    g = theta1 + y * theta2
+
+    dy_dt[:, 0] = -y[:, -2] * y[:, -1] + y[:, -1] * y[:, 1] - y[:, 0] + f - g[:, 0] + eta[:, 0]
+
+    dy_dt[:, 1] = -y[:, -1] * y[:, 0] + y[:, 0] * y[:, 2] - y[:, 1] + f - g[:, 1] + eta[:, 1]
+
+    dy_dt[:, 2:-1] = (-y[:, :-3] * y[:, 1:-2] + y[:, 1:-2] * y[:, 3:] - y[:, 2:-1] + f - g[:, 2:-1]
+                      + eta[:, 2:-1])
+
+    dy_dt[:, -1] = (-y[:, -3] * y[:, -2] + y[:, -2] * y[:, 0] - y[:, -1] + f - g[:, -1]
+                    + eta[:, -1])
+
+    return dy_dt
+
+
+def runge_kutta_ode_solver(ode, time_step, y, params):
+    """4th order Runge-Kutta ODE solver.
+
+    Carnahan, B., Luther, H. A., and Wilkes, J. O. (1969).
+    Applied Numerical Methods. Wiley, New York.
+
+    Parameters
+    ----------
+    ode : function
+        Ordinary differential equation function. In the Lorenz model it is SDE.
+    time_step : float
+    y : np.ndarray of dimension (batch_size, n_obs)
+        Current state of the time-series.
+    params : list of parameters
+        The parameters needed to evaluate the ode. In this case it is
+        list of four elements - eta, theta1, theta2 and f.
+
+    Returns
+    -------
+    np.ndarray
+        Resulting state initiated at y and satisfying ode solved by this solver.
+
+    """
+    k1 = time_step * ode(y, params)
+
+    k2 = time_step * ode(y + k1 / 2, params)
+
+    k3 = time_step * ode(y + k2 / 2, params)
+
+    k4 = time_step * ode(y + k3, params)
+
+    y = y + (k1 + 2 * k2 + 2 * k3 + k4) / 6
+
+    return y
+
+
+def forecast_lorenz(theta1=None, theta2=None, f=10., phi=0.3, n_obs=40, n_timestep=160,
+                    batch_size=1, initial_state=None, random_state=None, total_duration=4):
+    """Forecast Lorenz model.
+
+    Wilks, D. S. (2005). Effects of stochastic parametrizations in the
+    Lorenz ’96 system. Quarterly Journal of the Royal Meteorological Society,
+    131(606), 389–407.
+
+    Parameters
+    ----------
+    theta1, theta2: list or numpy.ndarray
+        Closure parameters. If the parameter is omitted, sampled from the prior.
+    phi : float, optional
+        This value is used to express stochastic forcing term. It should be configured according
+        to force term and eventually impacts to the result of eta.
+        More details in Wilks (2005) et al.
```

Comment:
In Wilks \phi is either 0 or close to 1, so where does the 0.3 used here come from?